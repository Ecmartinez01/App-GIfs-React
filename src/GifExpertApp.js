import React, {useState} from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';


export const GifExpertApp = ()=>{
    // const categorias = ["hya","wdsasd","sdawdw","asdadas"]
    const [categorias, setCategorias] = useState(['goku']);

    return (
         <>
          <h2 className='animate__animated animate__lightSpeedInLeft animate__delay-1s'>GifExpertApp</h2>
          <AddCategory setCategory={setCategorias}/>
          <hr/>

          <ol>
              {
                  categorias.map(categ =>(
                      <GifGrid key={categ} 
                      categoria={categ}
                      />
                  ))
              }     
          </ol>
          </>
    )
}