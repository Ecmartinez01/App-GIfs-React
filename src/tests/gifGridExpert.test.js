import { shallow } from "enzyme"

import React from 'react';
import { GifExpertApp } from "../GifExpertApp";

describe('Pruebas con gifGridExpert', () => {
    it('Dederia mostrar el copmponente correctamente', () => {
        const wrapper = shallow(<GifExpertApp/>)
        expect(wrapper).toMatchSnapshot()
    })
    it('debe mostrar una lista de coategorias', () => {
        const defaultCategory = ["juan","pepe"]
        const wrapper = shallow(<GifExpertApp defaultCategory={defaultCategory}/>)
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find("GifGrid").length).toBe(defaultCategory.length)
    })
     
    
})
