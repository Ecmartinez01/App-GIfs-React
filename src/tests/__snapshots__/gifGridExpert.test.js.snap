// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con gifGridExpert Dederia mostrar el copmponente correctamente 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__lightSpeedInLeft animate__delay-1s"
  >
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas con gifGridExpert debe mostrar una lista de coategorias 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__lightSpeedInLeft animate__delay-1s"
  >
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="juan"
      key="juan"
    />
    <GifGrid
      categoria="pepe"
      key="pepe"
    />
  </ol>
</Fragment>
`;
